### ISSUE TITLE
False negative: `ARG002` is not triggered for methods decorated with `typing_extensions.override`

### ISSUE THREAD

**Issue by bryanforbes** [charliermarsh/ruff#3952]
Ruff version: 0.0.261

Given the following file `foo.py`:

```python
from __future__ import annotations

from typing_extensions import override


class Foo:
    def do_something(self, arg: int) -> None:
        print('here')


class Bar(Foo):
    @override
    def do_something(self, arg: int) -> None:
        print('here as well')
```

And the following invocation:

```sh
ruff --select ARG --isolated --no-cache foo.py
```

The following output is produced:

```
foo.py:7:28: ARG002 Unused method argument: `arg`
```

There should be another error reported at line 13. For reference, [PEP-698](https://peps.python.org/pep-0698/) defines `@override`.

**Comment by charliermarsh** [comment]
I believe this is intentional, the reason being that if you're overriding a method, you can't really control the signature. For the same reason, we ignore unused arguments in abstract methods. It looks like these are configuration options in the originating plugin ([flake8-unused-arguments](https://pypi.org/project/flake8-unused-arguments/)).

What would you consider to be the right thing to do here, as a user? Underscore-prefix the variable name?

**Comment by bryanforbes** [comment]
> I believe this is intentional, the reason being that if you're overriding a method, you can't really control the signature.

This makes sense and I should have thought of that. Let's blame it on all the meetings I was in yesterday.

> It looks like these are configuration options in the originating plugin ([flake8-unused-arguments](https://pypi.org/project/flake8-unused-arguments/)).

It might be nice to give users the option to do this, but after reading that first sentence, I'm not sure that's even needed.

> What would you consider to be the right thing to do here, as a user? Underscore-prefix the variable name?

For clarity's sake, the original method I was working with had a positional-only argument. If we take a look at all of the types of arguments:

```python
class Foo:
    def do_something(
        self, arg: int, /, arg2: str, *, arg3: str, arg4: str | None = None
    ) -> None:
        print('here')


class Bar(Foo):
    @override
    def do_something(
        self, arg: int, /, arg2: str, *, arg3: str, arg4: str | None = None
    ) -> None:
        print('here as well')
```

In this case, `arg` could be renamed since it can never be called as a named argument, but none of the rest can be since they can all be called as named arguments. Positional-only arguments are the only case I can think of where I could see ruff flagging an argument as unused in an overridden method since they can be renamed.

**Comment by bryanforbes** [comment]
> To add to the discussion:
> https://peps.python.org/pep-0698/#no-new-rules-for-override-compatibility explicitly doesn't mention anything about the overridden argument name. (yet, it may come in further PEPs)

@Avasam sorry for the second ping, but I forgot something in my last reply. The PEP doesn't need to mention anything about overridden argument names because that is covered by basic type checking for [method overrides](https://mypy.readthedocs.io/en/stable/error_code_list.html?highlight=liskov#check-validity-of-overrides-override). If a type checker isn't already checking that before it implements PEP-698, it's not doing its job.

**Comment by charliermarsh** [comment]
Yeah this all makes sense -- I follow what you're saying RE positional-only arguments. I think the current behavior is reasonable, I think it would be surprising to users that positional-only arguments are flagged while all others aren't, even if it'd actually be totally fine to rename them. I'm gonna close but always open to follow-up questions :)
